### PR TITLE
JENA-1594: Per-graph access control for Fuseki

### PR DESCRIPTION
**Pull request by afs**
This adds a new module `jena-fuseki-access` that contains code for filtering data in SPARQL Query, SPARQL Graph Store Protocol, and plain quads access. Different users will see different views of the data based on their access rights.

It works with embedded Fuseki and uses Jetty for authentication. It would be possible to do the same for a general webapp container without direct Jetty dependencies. This is not planned for at this stage.

Also includes some smaller things picked up during development and tested as part 
of this development:

* [JENA-1585](https://issues.apache.org/jira/browse/JENA-1585) -- More refactoring of Fuseki for future separation of core engine and the webapp form of Fuseki.
* [JENA-1595](https://issues.apache.org/jira/browse/JENA-1595) -- log4j in scope test
* [JENA-1592](https://issues.apache.org/jira/browse/JENA-1592) -- Bug fix for TDB2
* Typos and clearing up.

### CODE REVIEW

**Comment by afs** [comment]
This PR is not yet completely finished. It is put out now to give it early visibility.

I have some development checking to do, some code tidying and documentation but the tests are reasonably comprehensive and do pass.

There is still work to be done to expose this conveniently on the command line. At the moment, it is a code library and needs java code to use.

File: jena-arq/src/main/java/org/apache/jena/sparql/core/DatasetGraphFilteredView.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  */
+
+    private final Predicate<Quad> quadFilter;
+    private final Collection<Node> visibleGraphs;
+
+    public DatasetGraphFilteredView(DatasetGraph dsg, Predicate<Quad> filter, Collection<Node> visibleGraphs) {
+        super(dsg);
+        this.quadFilter = filter;
+        if ( visibleGraphs.contains(Quad.defaultGraphIRI) || visibleGraphs.contains(Quad.defaultGraphNodeGenerated) ) {
+            Log.warn(DatasetGraphFilteredView.class, "default graph Node in visibleGraphs colelction - fix up applied");
```
Comment (rvesse) [inline jena-arq/src/main/java/org/apache/jena/sparql/core/DatasetGraphFilteredView.java]: Typo -> `colelction`

File: jena-fuseki2/jena-fuseki-access/src/main/java/org/apache/jena/fuseki/access/DataAccessCtl.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import org.apache.jena.query.DatasetFactory;
+import org.apache.jena.riot.WebContent;
+import org.apache.jena.sparql.core.DatasetGraph;
+import org.apache.jena.sparql.core.DatasetGraphFilteredView;
+import org.apache.jena.sparql.util.Context;
+import org.apache.jena.sparql.util.Symbol;
+import org.apache.jena.sys.JenaSystem;
+import org.eclipse.jetty.security.SecurityHandler;
+
+/** A library of operations related to data acess sexurity for Fuseki */
```
Comment (rvesse) [inline jena-fuseki2/jena-fuseki-access/src/main/java/org/apache/jena/fuseki/access/DataAccessCtl.java]: Typo -> `sexurity`

**Comment by afs** [comment]
Thanks - fixed applied.

**Pull request merged**